src/aws/bin/terraform

```diff
@@ -2,12 +2,54 @@
 
 set -eo pipefail
 
-# Run Terraform commands in the docker container passing our environment variables
-
-docker run --rm -it \
-    -u "$(id -u)" \
-    -v "${PWD}/:/app" \
-    -w "/app" \
-    --env-file ./env.d/development \
-    "hashicorp/terraform:0.14.5" \
-    "$@"
+declare MARSHA_TERRAFORM_ENV_FILE
+
+function terraform() {
+    current_workspace=$(
+        docker run --rm -i \
+            -u "$(id -u)" \
+            -v "${PWD}/:/app" \
+            -w "/app" \
+            "hashicorp/terraform:0.14.5" \
+            workspace show
+    ) 
+
+    # If an environment file is specified, it must match the current workspace
+    if [[ -n ${MARSHA_TERRAFORM_ENV_FILE+x} ]]; then
+        if [[ ${current_workspace} != "${MARSHA_TERRAFORM_ENV_FILE}" ]]; then
+            echo -e "workspace ${current_workspace} does not match environment file ${MARSHA_TERRAFORM_ENV_FILE}"
+            exit 1
+        fi
+    else
+        # Else we fallback to the developement file and the current workspace must start with "dev-"
+        if [[ ! ${current_workspace} =~ ^dev-.*$ ]]; then
+            echo "workspace used must start by \"dev-\""
+            exit 1
+        fi
+        
+        MARSHA_TERRAFORM_ENV_FILE="development"
+    fi
+
+    docker run --rm -it \
+        -u "$(id -u)" \
+        -v "${PWD}/:/app" \
+        -w "/app" \
+        --env-file ./env.d/${MARSHA_TERRAFORM_ENV_FILE} \
+        "hashicorp/terraform:0.14.5" \
+        "$@"
+}
+
+OPTS=$(getopt -o "e" --long "env:" -n "arnold" -- "$@")
+eval set -- "$OPTS"
+# Parse options to the `arnold` command
```

Comment:
copy/pasta fail?

Reply:
How did you notice it ?

Reply:
There is still an "arnold" in the command :stuck_out_tongue:

Reply:
rofl...

Reply:
Done